core/src/main/java/com/sequenceiq/cloudbreak/service/upgrade/image/PreWarmParcelLocationFilter.java

```diff
@@ -58,41 +61,50 @@ private boolean isInvalidImage(Image image) {
     private boolean isEligibleForUpgrade(Image image, Set<String> requiredParcels) {
         Map<String, Set<String>> preWarmParcelsByParcelName = getPreWarmParcelsByParcelName(image, requiredParcels);
         LOGGER.debug("Available pre warm parcels: {} on image: {}", preWarmParcelsByParcelName, image.getUuid());
-        return !preWarmParcelsByParcelName.isEmpty() && allRequiredPreWarmParcelAvailable(preWarmParcelsByParcelName)
-                && allRequiredPreWarmParcelUrlLocationEligible(preWarmParcelsByParcelName);
+        return allAvailableRequiredPreWarmParcelHasValue(preWarmParcelsByParcelName)
```

Comment:
same as above

Reply:
`return requiredParcels.isEmpty() || isEligibleForUpgrade(image, requiredParcels);` covers this